src/Proxy.hs

```diff
@@ -1999,22 +2014,22 @@ getNextPaydayR =
       sendResponse $ toJSON timestampObject
   where
     doGetParameters = do
-      rewardStatus <- withLastFinalBlockHash Nothing getRewardStatus
-      return $ do
-        response <- rewardStatus
-        utc <- parseEither
-          ( withObject "Best finalized block" $ \v -> do
-              v AE..: "nextPaydayTime"
-          ) $ grpcResponseVal response
-        Right $ GRPCResponse (grpcHeaders response) utc
+      rewardStatusRes <- getTokenomicsInfo Best
+      case getResponseValueAndHeaders rewardStatusRes of
+        Left errRes -> return errRes
+        Right (Left err, hds) -> return $ StatusOk $ GRPCResponse hds $ Left err
+        Right (Right rewardStatus, hds) -> do
+          case rewardStatus of
+            RewardStatusV0{} -> -- This should not happen
```

Comment:
I don't see why not. If the chain is on P3, why would this not happen?

It's fine to raise an error here, but the error is should be about invalid request or invalid state. It is not a GRPC error.